app/src/main/java/io/gnosis/safe/ui/settings/safe/SafeSettingsFragment.kt

```diff
@@ -106,25 +108,39 @@ class SafeSettingsFragment : BaseViewBindingFragment<FragmentSettingsSafeBinding
     private fun showSafeInfo(
         safe: Safe?,
         safeInfo: SafeInfo?,
-        ensNameValue: String?
+        ensNameValue: String?,
+        localOwners: List<Owner>
     ) {
         with(binding) {
             localName.name = safe?.localName
             threshold.name = getString(R.string.safe_settings_confirmations_required, safeInfo?.threshold, safeInfo?.owners?.size)
             ownersContainer.removeAllViews()
-            safeInfo?.owners?.forEach { owner -> ownersContainer.addView(ownerView(owner)) }
-            masterCopy.setAddress(safeInfo?.masterCopy)
+            safeInfo?.owners?.forEach { owner -> ownersContainer.addView(ownerView(owner.value, localOwners)) }
+            masterCopy.setAddress(safeInfo?.implementation?.value, safeInfo?.version)
             ensName.name = ensNameValue?.takeUnless { it.isBlank() } ?: getString(R.string.safe_settings_not_set_reverse_record)
             mainContainer.visible(true)
         }
     }
 
-    private fun ownerView(owner: Solidity.Address): AddressItem {
-        return AddressItem(requireContext()).apply {
-            background = ContextCompat.getDrawable(requireContext(), R.drawable.background_selectable_white)
-            layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, resources.getDimension(R.dimen.item_address).toInt())
-            address = owner
-            showSeparator = true
+    private fun ownerView(owner: Solidity.Address, localOwners: List<Owner>): View {
+
+        val localOwner = localOwners.find { it.address == owner }
+
+        return if (localOwner != null) {
+            NamedAddressItem(requireContext()).apply {
+                background = ContextCompat.getDrawable(requireContext(), R.drawable.background_selectable_white)
+                layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, resources.getDimension(R.dimen.item_address).toInt())
+                address = owner
+                name = localOwner.name
```

Comment:
ATM this seems to be the null for the imported owner. Can we show "Imported owner key" instead? Until we have multiple owner names?

Reply:
Right now it looks like this: 
![image](https://user-images.githubusercontent.com/246473/111526189-b8adb700-875e-11eb-869d-e8bbbba44299.png)

Reply:
yes, this is how it is actually defined in spec. Thank you for pointing it out